db-engine/src/main/java/com/splicemachine/db/iapi/types/SqlXmlUtil.java

```diff
@@ -684,7 +684,7 @@ protected List evalXQExpression(XMLDataValue xmlContext,
          * XML_DOC_ANY (which means we can store it in a Derby
          * XML column).
          */
-        if ((itemRefs.size() == 1) && (itemRefs.get(0) instanceof Document))
```

Comment:
I think it'd be better to raise an exception in the default: case where we set itemRefs to null. With this change we silently continue processing after not knowing how to handle some XPathResult type.

Reply:
Committed new one that thrown an exception for unknown XPath result type. Thanks.